R/fit.R

```diff
@@ -953,7 +998,24 @@ CmdStanMCMC <- R6::R6Class(
         out <- lapply(out, diag)
       }
       out
+    },
+    # approximate loo-cv using the loo package
+    loo = function(variables = "log_lik", r_eff = TRUE, ...) {
+      if (!requireNamespace("loo", quietly = TRUE)) {
+        stop("Please install the loo package to use this method.", call. = FALSE)
+      }
+      LLarray <- self$draws(variables)
+      if (is.logical(r_eff)) {
+        if (isTRUE(r_eff)) {
+          r_eff_cores <- list(...)[["cores"]] %||% getOption("mc.cores", 1)
+          r_eff <- loo::relative_eff(exp(LLarray), cores = r_eff_cores)
+        } else {
+          r_eff <- NULL
+        }
+      }
+      loo::loo.array(LLarray, r_eff = r_eff, ...)
```

Comment:
@rok-cesnovar @avehtari Right now the loo method just returns the loo object, but we could also store the results in the fit object so that repeated calls to loo don't need to do the computation again (perhaps unless the user selects an option to force it). That could look roughly like this: 

```r
loo = function(..., force = FALSE) {
  ...
  if (is.null(private$loo_object_) || force) {
    private$loo_object_ <- loo::loo.array(LLarray, r_eff = r_eff, ...)
  }
  return(private$loo_object_)
}
```

I'm not sure if we want to do this or not. On the one hand it's pretty convenient, on the other hand users can just assign the result of `loo$fit()` to some object to avoid recomputing so maybe we don't need to do this. What do you think?

Reply:
I think storing it separately is the way to go.

Reply:
Ok I agree, let's not try to get too fancy here.

Reply:
btw. brms has add_criterion function for adding to the fit object, but we don't need to do that here or now